### DISCUSSION TITLE
exception handling

### DISCUSSION THREAD

**Discussion by hamidrasti** [NilCoalescing/djangochannelsrestframework#142] [FAQ]
How to handle exceptions in a proper way?

**Comment by hishnash** [comment]
What do you want to do with exceptions?  Currently if your `@action` methods throw then a message is sent down the channel to the client it should include etc request ID and information from the execution using the same info as RestFramework.  

If you want to customise this you can override the   `handle_exception` method https://github.com/NilCoalescing/djangochannelsrestframework/blob/master/djangochannelsrestframework/consumers.py#L117-L136

**Reply by hamidrasti** [reply]
What if the exception raised in async `connect` method? How can we handle that? 
Thanks in advance for your help.

**Reply by hishnash** [reply]
sorry for the slow response.

if an exception is rased in the connect method then the connection will be closed by Django Channels.  I believe it is not possible to send a response to the client unless the connect method completes.
